aten/src/ATen/native/sparse/cuda/SparseCUDATensorMath.cu

```diff
@@ -511,6 +518,52 @@ SparseTensor& mul_out_sparse_cuda(const SparseTensor& t_, const SparseTensor& sr
   return r_._coalesced_(true);
 }
 
+// --------------------------------------------------------------------
+// mul(Tensor, SparseTensor)
+// --------------------------------------------------------------------
+
+SparseTensor& mul_out_sparse_dense_cuda(SparseTensor& r, const SparseTensor& sparse_, const Tensor& dense) {
```

Comment:
This function appears to be identical to `mul_out_sparse_dense_cpu` when not looking at checks containing `is_cuda` and noticing that `cuda::check_device` is unnecessary as it is already handled in `mul_out_sparse_cuda`.

Suggestion: rename `mul_out_sparse_dense_cpu` to `mul_out_sparse_dense` and use it both in `mul_out_sparse_cpu` and `mul_out_sparse_cuda`.